src/mrinufft/trajectories/inits/travelling_salesman.py

```diff
@@ -95,15 +98,15 @@ def _cluster_by_coordinate(
 def _initialize_ND_travelling_salesman(
     Nc: int,
     Ns: int,
-    density: np.typing.NDArray,
-    first_cluster_by: Literal | None = None,
-    second_cluster_by: Literal | None = None,
-    sort_by: Literal | None = None,
+    density: NDArray,
+    first_cluster_by: Literal["x", "y", "z", "r", "phi", "theta"] | None = None,
+    second_cluster_by: Literal["x", "y", "z", "r", "phi", "theta"] | None = None,
+    sort_by: Literal["x", "y", "z", "r", "phi", "theta"] | None = None,
```

Comment:
I think you can:

- Use a type alias (declared below the imports), for brievety
- Put the None inside the literal